bgpd/bgp_nb_config.c

```diff
@@ -332,33 +669,62 @@ int routing_control_plane_protocols_control_plane_protocol_bgp_global_route_refl
 int routing_control_plane_protocols_control_plane_protocol_bgp_global_route_reflector_allow_outbound_policy_modify(
 	struct nb_cb_modify_args *args)
 {
-	switch (args->event) {
-	case NB_EV_VALIDATE:
-	case NB_EV_PREPARE:
-	case NB_EV_ABORT:
-	case NB_EV_APPLY:
-		/* TODO: implement me. */
-		break;
-	}
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	struct bgp *bgp;
+	int ret;
+
+	bgp = nb_running_get_entry(args->dnode, NULL, true);
+
+	if (yang_dnode_get_bool(args->dnode, NULL))
+		SET_FLAG(bgp->flags, BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY);
+	else
+		UNSET_FLAG(bgp->flags, BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY);
+
+	update_group_announce_rrclients(bgp);
+
+	ret = bgp_clear_star_soft_out(bgp->name, args->errmsg,
+				      args->errmsg_len);
+	if (ret != CMD_SUCCESS)
+		return NB_ERR_INCONSISTENCY;
 
 	return NB_OK;
 }
 
+/*
+ * XPath:
+ * /frr-routing:routing/control-plane-protocols/control-plane-protocol/frr-bgp:bgp/global/route-selection-options
+ */
+void routing_control_plane_protocols_control_plane_protocol_bgp_global_route_selection_options_apply_finish(
+	struct nb_cb_apply_finish_args *args)
+{
+	struct bgp *bgp;
+
+	bgp = nb_running_get_entry(args->dnode, NULL, true);
+
+	bgp_recalculate_all_bestpaths(bgp);
```

Comment:
Not related to this PR as VTY is also doing the same but I think we have some kind of optimization that can be done here. If any of the best path selection flags did not change then we should not call bgp_recalculate_all_bestpaths or set some flag in BGP such that inside bgp_recalculate_all_bestpaths do not best path selection again.